packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -455,6 +612,19 @@ abstract class ImageStreamCompleter with Diagnosticable {
     }
   }
 
+  /// Calls all the registered listeners to notify them of a new image.
+  @protected
+  void setImage(ImageInfo image) {
+    _checkDisposed();
+    _currentImage?.image.dispose();
+    _currentImage = image;
+
+    if (_activeListeners.isEmpty)
+      return;
+    _notify(image, _activeListeners);
+    _notify(image, _passiveListeners);
```

Comment:
Why are passive listeners not informed here? The doc for them includes:

> but does want to know the latest frame used in case new clients ask for the cached image.

Wouldn't the cache then have an outdated image?

Reply:
It will get the updated image once an active listener registers, as will the active listener.

The cache itself doesn't actually need the image reference, it just needs the completer to give out.